javarosa/src/main/java/org/javarosa/form/api/FormEntryController.java

```diff
@@ -369,4 +374,77 @@ public void setLanguage(String language) {
     public String getFormEntrySessionString() {
         return formEntrySession.toString();
     }
+    
+    /**
+     * getQuestionPrompts for the current index
+     */
+    public FormEntryPrompt[] getQuestionPrompts() throws RuntimeException {
+        return getQuestionPrompts(getModel().getFormIndex());
+    }
+
+    /**
+     * Returns an array of relevant question prompts that should be displayed as a single screen.
+     * If the given form index is a question, it is returned. Otherwise if the
+     * given index is a field list (and _only_ when it is a field list)
+     */
+    public FormEntryPrompt[] getQuestionPrompts(FormIndex currentIndex) throws RuntimeException {
+
+        IFormElement element = this.getModel().getForm().getChild(currentIndex);
+
+        //If we're in a group, we will collect of the questions in this group
+        if (element instanceof GroupDef) {
+            //Assert that this is a valid condition (only field lists return prompts)
+            if (!this.isFieldListHost(currentIndex)) {
+                throw new RuntimeException("Cannot get question prompts from a non-field-list group");
+            }
+
+            // Questions to collect
+            ArrayList<FormEntryPrompt> questionList = new ArrayList<>();
+
+            //Step over all events in this field list and collect them
+            FormIndex walker = currentIndex;
+
+            int event = this.getModel().getEvent(currentIndex);
+            while (FormIndex.isSubElement(currentIndex, walker)) {
+                if (event == FormEntryController.EVENT_QUESTION) {
+                    questionList.add(this.getModel().getQuestionPrompt(walker));
+                }
+
+                if (event == FormEntryController.EVENT_PROMPT_NEW_REPEAT) {
+                    //TODO: What if there is a non-deterministic repeat up in the field list?
+                }
+
+                //this handles relevance for us
+                walker = this.getNextIndex(walker);
+                event = this.getModel().getEvent(walker);
+            }
+
+            FormEntryPrompt[] questions = new FormEntryPrompt[questionList.size()];
+            //Populate the array with the collected questions
+            questionList.toArray(questions);
+            return questions;
+        } else {
+            // We have a question, so just get the one prompt
+            return new FormEntryPrompt[]{this.getModel().getQuestionPrompt(currentIndex)};
+        }
+    }
+
+    /**
+     * A convenience method for determining if the current FormIndex is a group that is/should be
+     * displayed as a multi-question view of all of its descendants. This is useful for returning
+     * from the formhierarchy view to a selected index.
+     */
+    private boolean isFieldListHost(FormIndex index) {
```

Comment:
This method is now duplicated in `FormController` and `FormEntryController`, right? Can you make the implementation in core static and have the android code call that?